### ISSUE TITLE
Check Spectre Variant 1 on ARM 32  

### ISSUE THREAD

**Issue by bprout93** [speed47/spectre-meltdown-checker#225]
Is there any way to check Spectre Variant 1 on ARM 32 bit processor ? Now I am using kernel version 4.14 and applied spectre variant 1 patch for arm. Patch link:http://git.arm.linux.org.uk/cgit/linux-arm.git/log/?h=spectre

**Comment by speed47** [comment]
There is, by detecting the array_index_mask_nospec macro usage. Working on it, should be available soon, maybe tonight if I don't encounter any issue

**Comment by bprout93** [comment]
Yes, I am running the script on real ARM system, But there is is no detection of "csdb" instruction after running below command :

#  objdump -d vmlinux | grep -iw csdb

I have applied below three patches for Spectre-v1 on kernel-4.14.52
1. ARM: spectre-v1: fix syscall entry
2. ARM: spectre-v1: add array_index_mask_nospec() implementation
3. ARM: spectre-v1: add speculation barrier (csdb) macros

Do you think these patches will mitigate the Spectre Variant 1 issue on ARM 32 bit or any other patches required ?

**Comment by speed47** [comment]
This is normal, csdb is seen for ARM64 instructions, not for ARM32.

Did you try the new arm branch ? (https://raw.githubusercontent.com/speed47/spectre-meltdown-checker/master/spectre-meltdown-checker.sh), it should detect the ARM32 case now, in the "Kernel has array_index_mask_nospec" test

**Comment by speed47** [comment]
This has been included in the v0.38 release, to avoid delaying it further.
Please comment if you can confirm it works as expected under a native ARM 32 system (not arm64)

**Comment by bprout93** [comment]
I have tested the script with vmlinux raw ELF file and it detects the "array_index_mask_nospec" on arm32 native system.But the script fails to detect in case of compressed kernel image such as zImage and uImage. Is there any way to decompress zImage to vmlinux raw ELF file.

**Comment by speed47** [comment]
@bprout93 if you still have said system around, I've added an experimental way of detecting raw binary arm images and adjusting `objdump`'s options when this is the case. Could you try the `arm64` branch? (there are other commits for arm64 in this branch, but the last commit is actually for arm32)
